storage/aptosdb/src/lib.rs

```diff
@@ -1343,6 +1347,40 @@ impl DbWriter for AptosDB {
                 .get_snapshot_receiver(version, expected_root_hash)
         })
     }
+
+    fn finalize_state_snapshot(
+        &self,
+        version: Version,
+        output_with_proof: TransactionOutputListWithProof,
+    ) -> Result<()> {
+        // Update the merkle accumulator using the given proof
```

Comment:
All I ask is to assert `output_with_proof` is of length 1, at this point.
And document it in the interface.

Better if we pass in types that imply that.. :D